core/src/main/java/com/sequenceiq/cloudbreak/conf/RoutingDataSource.java

```diff
@@ -0,0 +1,42 @@
+package com.sequenceiq.cloudbreak.conf;
+
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.datasource.lookup.AbstractRoutingDataSource;
+
+public class RoutingDataSource extends AbstractRoutingDataSource {
+
+    public static final String QUARTZ_THREAD_NAME_PREFIX = "quartzscheduler";
+
+    private static final String DEFAULT_DATASOURCE_KEY = "default";
+
+    private static final String QUARTZ_DATASOURCE_KEY = "quartz";
+
+    @Inject
+    @Qualifier("defaultDataSource")
+    private DataSource defaultDataSource;
+
+    @Inject
+    @Qualifier("quartzDataSource")
+    private DataSource quartzDataSource;
+
+    @PostConstruct
+    void setUp() {
+        setTargetDataSources(Map.of(
+                DEFAULT_DATASOURCE_KEY, defaultDataSource,
+                QUARTZ_DATASOURCE_KEY, quartzDataSource
+        ));
+        setDefaultTargetDataSource(defaultDataSource);
+    }
+
+    @Override
+    protected Object determineCurrentLookupKey() {
+        boolean quartzThread = Thread.currentThread().getName().toLowerCase().startsWith(QUARTZ_THREAD_NAME_PREFIX);
```

Comment:
Is there a more explicit (rather than using the thread name) way to figure out if we are inside a quartz job?

Reply:
Well we could also set a ThreadLocal value when we are in a quartz job, and check for it here, but that does not seem much better to me. Would you prefer that, or do you have any better ideas?

Reply:
Yes not much of a difference, I fine with this.